frontend/components/site/siteBuildsBuild.jsx

```diff
@@ -0,0 +1,327 @@
+    switch (state) {
+      case 'error':
+        messageStatusDoneIcon = {
+          messagePrefix: 'Failed in ',
+          status: error === 'The build timed out' ? 'Timed out' : 'Failed',
+          done: true,
+          icon: IconExclamationCircle,
+        };
+        break;
+      case 'processing':
+        messageStatusDoneIcon = {
+          messagePrefix: 'In progress for ',
+          status: 'In progress',
+          done: false,
+          icon: IconSpinner,
+        };
+        break;
+      case 'skipped':
+        messageStatusDoneIcon = {
+          messagePrefix: 'Skipped',
+          status: 'Skipped',
+          done: true,
+          icon: IconX,
+        };
+        break;
+      case 'queued':
+      case 'tasked':
+      case 'created':
+        messageStatusDoneIcon = {
+          messagePrefix: 'In queue for ',
+          status: 'Queued',
+          done: false,
+          icon:
+          IconClock,
+        };
+        break;
+      case 'success':
+        messageStatusDoneIcon = {
+          messagePrefix: 'Done in ',
+          status: 'Completed',
+          done: true,
+          icon: IconCheckCircle,
+        };
+        break;
+      default:
+        messageStatusDoneIcon = {
+          messagePrefix: state,
+          status: state,
+          done: null,
+          icon: null,
+        };
+    }
+    return messageStatusDoneIcon;
+  };
+
+  const {
+    messagePrefix, status, done, icon,
+  } = buildStateData(build);
+
+  let assumeScanQueued = false;
+
+  function buildLogsLink(buildId) {
+    const cta = 'View build logs';
+    return <Link to={`/sites/${siteId}/builds/${buildId}/logs`}>{cta}</Link>;
+  }
+
+  function shaLink(thisBuild) {
+    const { owner, repository } = thisBuild.site;
+    const sha = thisBuild.clonedCommitSha || thisBuild.requestedCommitSha;
+    if (sha) {
+      return (
+        <GitHubLink
+          owner={owner}
+          repository={repository}
+          sha={sha}
+          branch={null}
+          text={sha.slice(0, 7)}
+          icon="sha"
+        />
+      );
+    }
+    return null;
+  }
+
+  function branchLink(thisBuild) {
+    const { owner, repository } = thisBuild.site;
+
+    return (
+      <GitHubLink
+        owner={owner}
+        repository={repository}
+        sha={null}
+        branch={thisBuild.branch}
+        text={thisBuild.branch}
+        icon="branch"
+      />
+    );
+  }
+
+  function isPreviewBuild(thisBuild) { return previewBuilds[thisBuild.branch] === thisBuild.id; }
+
+  function buildHasBuildTasks(thisBuild) {
+    return thisBuild.BuildTasks?.length > 0;
+  }
+
+  function buildTasksInProgress(thisBuild) {
+    return thisBuild.BuildTasks?.length && thisBuild.BuildTasks.every(task => !!(task.status !== 'success') || (task.status !== 'error'));
+  }
+
+  const summarizeTaskResults = (({ BuildTasks }, scanWasQueued = false) => {
+    const anyTaskErrored = BuildTasks.find(task => task.status === 'error');
```

Comment:
When running the rebuild on a site build, the frontend errors when trying to call `.find`. I believe it could be coming from  the here.
<img width="387" alt="• React Router caught the following" src="https://github.com/user-attachments/assets/393e11b7-7286-4b94-a1d7-6b7646ac52a8">